main.md

```diff
@@ -875,24 +875,29 @@ prolonged deployments of protected resources with mixed token type support.
 
 # Authorization Server-Provided Nonce {#ASNonce}
 
-Including a nonce value contributed by the authorization server in the DPoP proof
-MAY be used by authorization servers to limit the lifetime of DPoP proofs.
-The server is in control of when to require the use of a new nonce value
-in subsequent DPoP proofs.
-
-Without employing such a mechanism, a malicious party controlling the client
+A malicious party controlling the client
 (including potentially the end user)
 can create DPoP proofs for use arbitrarily far in the future.
-This section specifies how server-provided nonces are used with DPoP.
+This section specifies a mechanism based on opaque nonces provided by the authorization server
+that can be used to limit the lifetime of DPoP proofs.
+
+The authorization server decides when to require the use of a new nonce value
+in subsequent DPoP proofs.
+
+When an authorization server wants the client to include a nonce in the DPoP proof,
+it replies to requests without a nonce with a specific error response conveying the nonce.
+The response:
+
+* MUST convey the nonce in the `DPoP-Nonce` HTTP header field;
+* MUST NOT convey more than one `DPoP-Nonce` HTTP header field;
+* have HTTP status `400` (Bad Request) per Section 5.2 of [@!RFC6749] using `use_dpop_nonce` as the
+error code value.
 
-An authorization server MAY supply a nonce value to be included by the client
-in DPoP proofs sent. In this case, the authorization server responds to requests not including a nonce
-with an HTTP `400` (Bad Request) error response per Section 5.2 of [@!RFC6749] using `use_dpop_nonce` as the
-error code value. The authorization server includes a `DPoP-Nonce` HTTP header in the response supplying
-a nonce value to be used when sending the subsequent request.
 This same error code is used when supplying a new nonce value when there was a nonce mismatch.
-The client will typically retry the request with the new nonce value supplied
-upon receiving a `use_dpop_nonce` error with an accompanying nonce value.
+
+A compliant client receiving such a response,
+MUST retry the request supplying the nonce encoding its value in the `nonce` claim
+of the DPoP proof.
```

Comment:
No, it's not a MUST for the client to retry.  The client is always free to abandon the interaction with the server at any time.  I know what you're trying to say, but please reword such that you're not precluding abandonment by the client, which is always an option.